c10/util/Optional.cpp

```diff
@@ -11,4 +12,10 @@ static_assert(
 static_assert(
     C10_IS_TRIVIALLY_COPYABLE(c10::optional<bool>),
     "c10::optional<bool> should be trivially copyable");
+static_assert(
+    C10_IS_TRIVIALLY_COPYABLE(c10::optional<c10::IntArrayRef>),
+    "c10::optional<IntArrayRef> should be trivially copyable");
+static_assert(
+    sizeof(c10::optional<c10::IntArrayRef>) == sizeof(c10::IntArrayRef),
+    "c10::optional<IntArrayRef> should be size-optimized");
```

Comment:
This is assert is good, just shouldn't be in this file (because it's likely to get deleted when we go to std::optional)

Reply:
if people are going to delete this file without looking at it, what's going to stop them from deleting the contents of Optional.h or Optional_test.cpp without looking? (and which one of those two did you want this moved to)

Reply:
Blugh, IDK, some random test file in ATen/test maybe. Or we can just trust in code review, I guess.